feat (Codespaces) - Added Commonly used tasks to `tasks.json` for VSCode users.

ShreyasJejurkar opened the pull request:
Further, improving codespaces/VScode experience for users.

1. Added EditorConfig as a preinstalled extension. 
2. Added commonly performed tasks to the `tasks.json` file. So that one can invoke that task through VSCode directly by pressing `CTRL + SHIFT + P` and then select option `Run Tasks` and then actual tasks. (We can add more tasks going forward.)

Press `CTRL + SHIFT + P` and search for Run Tasks. Select option like below. 

![image](https://user-images.githubusercontent.com/17148381/139052595-9156cba0-d7c1-4708-85af-9d1475941804.png)


Choose the task from list and hit enter to execute. 

![image](https://user-images.githubusercontent.com/17148381/139052925-3703f871-5a8b-4165-8e11-ae588d036bb8.png)

Some tasks may have more options to select, for example in the below case while building an entire repository, one can select different configurations like Debug OR Release. And based on the selection the command will run. 

![image](https://user-images.githubusercontent.com/17148381/139053193-552433a9-f8f7-4bcb-883e-22fce344e212.png)

And given task will start running in the terminal. 

![image](https://user-images.githubusercontent.com/17148381/139053363-8cd3911d-024f-456b-8d76-15d24636d567.png)

ShreyasJejurkar committed:
feat (codespaces) - Add EditorConfig as pre-installed extension.

ShreyasJejurkar committed:
feat (codespaces/vscode) - Add commonly performed tasks to tasks.json

captainsafia reviewed (approved):
Looks good overall! Left a few notes inline.

Another set of tasks that might be good to add is build/test for a particular project. I wonder if there is a way to craft a task input that lists all the .csproj files in the repo...

ShreyasJejurkar committed:
Addressed PR feedback.
1. Added C# XML documentation comments extension to suggested extension.
2. Added some more Omnisharp options to devcontainer based on our omnisharp.json file
3. Renamed tasks to Run all test projects.

ShreyasJejurkar committed:
Removed extra space

ShreyasJejurkar committed:
Formatted the Json file using inbuilt vs code formatter!

ShreyasJejurkar committed:
indent issues

ShreyasJejurkar commented on .devcontainer/devcontainer.json:
```diff
@@ -19,12 +19,16 @@
-		"omnisharp.enableMsBuildLoadProjectsOnDemand": true
+		"omnisharp.enableMsBuildLoadProjectsOnDemand": true,
+    "omnisharp.enableRoslynAnalyzers": true,
+    "omnisharp.enableEditorConfigSupport": true,
```
Formatted with VSCode formatter for JSON!

ShreyasJejurkar committed:
Update build instruction

ShreyasJejurkar committed:
Keep 8 tabs as previosuly!

ShreyasJejurkar commented:
Hi @captainsafia, How do you edit the ResX file in VSCode? 
1. Raw XML editing.
2. Any extension you use?

If you use any extension for the editing, let me know we will add those to recommendations as well in docs and in `devcontainer.json` for streamline experience.

captainsafia commented:
>  How do you edit the ResX file in VSCode?

I haven't figured out a great workflow for this yet unfortunately. On the rare occasions where I do interact with ResX files, I usually end up editing the raw XML. :/

I'm not sure if there is a CLI tool for generating code from a ResX. That would help with at least part of the process.
